elastic/beats
# [Filebeat] Check content type when reading s3 files

When file name has `.gz` suffix but with text/plain content type, `newS3BucketReader` function will fail when using s3 input in Filebeat. Instead of simply checking file name, check the actual content type from the response and then decide how to build the new reader. 

## How to test it:
Upload file to an S3 bucket and change the file metadata property to test this PR:
<img width="1245" alt="Screen Shot 2020-01-07 at 7 48 59 AM" src="https://user-images.githubusercontent.com/14081635/71904079-af6c9780-3122-11ea-8cdf-c0c019fdd5fe.png">

Upload a test1.txt.gz file and change content type to `text/plain`, s3 input should still be able to read the file.

closes https://github.com/elastic/beats/issues/15225
